exporter/containerimage/writer.go

```diff
@@ -214,10 +237,51 @@ func (ic *ImageWriter) exportLayers(ctx context.Context, refCfg cacheconfig.RefC
 	return out, err
 }
 
+func (ic *ImageWriter) exportAttestations(ctx context.Context, s session.Group, desc *ocispecs.Descriptor, attestations ...exporter.Attestation) ([]InTotoAttestation, error) {
+	span, ctx := tracing.StartSpan(ctx, "export attestations")
+
+	eg, ctx := errgroup.WithContext(ctx)
+	attestationsDone := oneOffProgress(ctx, "exporting attestations")
+
+	out := make([]InTotoAttestation, len(attestations))
+
+	for i, attestation := range attestations {
+		mount, err := attestation.PredicateRef.Mount(ctx, true, s)
+		if err != nil {
+			return nil, err
+		}
+
+		lm := snapshot.LocalMounter(mount)
+		src, err := lm.Mount()
+		if err != nil {
+			return nil, err
+		}
+		defer lm.Unmount()
+
+		rawPredicate, err := os.ReadFile(path.Join(src, attestation.PredicatePath))
+		if err != nil {
+			return nil, err
+		}
```

Comment:
Is the `rawPredicate` in here always a JSON or could it be actual raw bytes sometimes that need to be encoded. Should there be a switch for that? Or maybe defined by predicatetype?

Reply:
I was thinking case like https://gist.github.com/tonistiigi/084e9c69b553af104e36f287fa332af7#file-attestations_in_rekor-txt-L33 . In there any data you provide is encoded to the `Data` field of the object.

Reply:
I can see how this seems like a convenient way to just attest anything, it just seems too meaningless as anything under that 'cosign.sigstore.dev/attestation/v1' predicate type could be really just anything. I think we should definitely descope  this kind of questionable use-cases from this initial PR.

Reply:
I've reworked this block so that we directly construct an `intoto.Predicate` here - this is an `interface{}` value, so we need to explicitly add a JSON decode here (which I thought was here before, but actually isn't, woops).